source/custom/nidaqmx_service.custom.cpp

```diff
@@ -93,6 +94,79 @@ int32 CVICALLBACK SetWfmAttrCallback(
   }
 }
 
+int32 CVICALLBACK SetAnalogWfmAttrCallback(
+    const uInt32 channel_index,
+    const char attribute_name[],
+    const int32 attribute_type,
+    const void* value,
+    const uInt32 value_size_in_bytes,
+    void* callback_data)
+{
+  return SetWfmAttrCallbackTemplated<RepeatedPtrField<DoubleAnalogWaveform>>(
+    channel_index, attribute_name, attribute_type, value, value_size_in_bytes, callback_data);
+}
+
+int32 CVICALLBACK SetDigitalWfmAttrCallback(
+    const uInt32 channel_index,
+    const char attribute_name[],
+    const int32 attribute_type,
+    const void* value,
+    const uInt32 value_size_in_bytes,
+    void* callback_data)
+{
+  return SetWfmAttrCallbackTemplated<RepeatedPtrField<::ni::protobuf::types::DigitalWaveform>>(
+    channel_index, attribute_name, attribute_type, value, value_size_in_bytes, callback_data);
+}
+
+template<typename RequestType>
+int32 ParseWaveformAttributeMode(const RequestType* request)
```

Comment:
Parsing usually refers to extracting data from a string. Consider calling this "GetWaveformAttributeMode".